### PR TITLE
Fix back end route controller example

### PR DESCRIPTION
**Pull request by aschempp**
I think the menu listener is invalid as well. @sheeep can you check if the code works?

### CODE REVIEW

File: docs/dev/guides/backend-routes.md [hunk 1]
```diff
@@ -26,27 +26,39 @@ and is configured through annotations.
...
 use Symfony\Component\HttpFoundation\Response;
+use Twig\Environment as TwigEnvironment;
 
 /**
- * @Route("/contao", defaults={
- *     "_scope" = "backend",
- *     "_token_check" = true,
- *     "_backend_module" = "my-modules"
- * })
+ * @Route("/contao//my-backend-route",
```
Comment (fiedsch) [inline docs/dev/guides/backend-routes.md]: The double slash is not intended, is it?

File: docs/dev/guides/backend-routes.md [hunk 2]
```diff
@@ -26,27 +26,39 @@ and is configured through annotations.
...
+    public function __construct(TwigEnvironment $twig)
+    {
+        $this->twig = $twig;
+    }
+    
     /**
-     * @Route("/my-backend-route", name="app.backend-route")
-     * @Template("my_backend_route.html.twig")
+     * @Route("", name="app.backend-route")
      */
```
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md]: This whole annotation block is not needed anymore, is it?

File: docs/dev/guides/backend-routes.md [hunk 3]
```diff
@@ -26,27 +26,39 @@ and is configured through annotations.
...
- *     "_scope" = "backend",
- *     "_token_check" = true,
- *     "_backend_module" = "my-modules"
- * })
+ * @Route("/contao//my-backend-route",
+ *     name="app.backend-route" 
+ *     defaults={
+ *         "_scope" = "backend",
+ *         "_token_check" = true,
+ *         "_backend_module" = "my-modules"
```
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md]:
```suggestion
 *         "_backend_module" = "my-module"
```

File: docs/dev/guides/backend-routes.md [hunk 4]
```diff
@@ -26,27 +26,39 @@ and is configured through annotations.
...
+use Twig\Environment as TwigEnvironment;
 
 /**
- * @Route("/contao", defaults={
- *     "_scope" = "backend",
- *     "_token_check" = true,
- *     "_backend_module" = "my-modules"
- * })
+ * @Route("/contao//my-backend-route",
+ *     name="app.backend-route"
```
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md]:
```suggestion
 *     name=App\Action\BackendAction::class,
```

File: docs/dev/guides/backend-routes.md (hunk 1 above)
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md]:
```suggestion
 * @Route("/contao/my-backend-route",
```

File: docs/dev/guides/backend-routes.md [hunk 5]
```diff
@@ -26,27 +26,39 @@ and is configured through annotations.
 // src/Controller/BackendController.php
 namespace App\Controller;
```
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md:27]:
```suggestion
namespace App\Action;
```

File: docs/dev/guides/backend-routes.md [hunk 6]
```diff
@@ -26,27 +26,39 @@ and is configured through annotations.
 // src/Controller/BackendController.php
```
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md:26]:
```suggestion
// src/Action/BackendAction.php
```

File: docs/dev/guides/backend-routes.md [hunk 7]
```diff
@@ -61,8 +73,8 @@ you need to set this to true, in order to get it to work.
 the current route in order to highlight the currently active node in the back end menu.
 More on this later.
 
-Be sure to have imported your bundles Controllers in your `routing.yml` *before* 
-the `ContaoCoreBundle` routes. 
+Be sure to have imported your bundles Controllers in your `routing.yml` *before*
```
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md:73]:
```suggestion
Be sure to have imported your bundles Controllers and Actions in your `routing.yml` *before*
```

**Review by fritzmg — CHANGES_REQUESTED** [review]
Thank you @aschempp :)

Imho we should advocate the "Action Domain Responder" (ADR) pattern, especially when using `__invoke`.

File: docs/dev/guides/backend-routes.md (hunk 5 above)
Comment (aschempp) [reply docs/dev/guides/backend-routes.md:27]: 👎  as far as I know, these are always `Controller` in Symfony

File: docs/dev/guides/backend-routes.md (hunk 4 above)
Comment (aschempp) [reply docs/dev/guides/backend-routes.md]: I have never seen a route with a class name, not sure this works (and is best practice?)

File: docs/dev/guides/backend-routes.md (hunk 5 above)
Comment (fritzmg) [reply docs/dev/guides/backend-routes.md:27]: » https://symfony.com/doc/current/controller/service.html#invokable-controllers

File: docs/dev/guides/backend-routes.md (hunk 1 above)
Comment (aschempp) [reply docs/dev/guides/backend-routes.md]: removed in 77a32c0

File: docs/dev/guides/backend-routes.md (hunk 4 above)
Comment (fritzmg) [reply docs/dev/guides/backend-routes.md]: Yes this works and is much more useful as you can then simply write

```php
$this->router->generate(BackendAction::class)
```

in PHP.

File: docs/dev/guides/backend-routes.md (hunk 5 above)
Comment (fritzmg) [reply docs/dev/guides/backend-routes.md:27]: Hm, I was sure that Symfony used `App\Action` there, but evidently they don't. Not sure where I got the recommendation from then.

There was a PR about it, but it did not get merged. https://github.com/symfony/symfony-docs/pull/8153

According to the discussion, Symfony is net yet promoting the ADR pattern within the new `Action` namespace.

Comment (fritzmg) [reply docs/dev/guides/backend-routes.md:27]: » https://github.com/contao/docs/pull/227

File: docs/dev/guides/backend-routes.md [hunk 8]
```diff
@@ -26,27 +26,39 @@ and is configured through annotations.
...
+use Twig\Environment as TwigEnvironment;
 
 /**
- * @Route("/contao", defaults={
- *     "_scope" = "backend",
- *     "_token_check" = true,
- *     "_backend_module" = "my-modules"
- * })
+ * @Route("/contao/my-backend-route",
+ *     name="app.backend-route"
```
Comment (fritzmg) [inline docs/dev/guides/backend-routes.md]:
```suggestion
 *     name=App\Controller\BackendController::class 
```

**Comment by aschempp** [comment]
is there any recommendation from Symfony on how to name routes?
/cc @contao/developers

File: docs/dev/guides/backend-routes.md (hunk 4 above)
Comment (richardhj) [reply docs/dev/guides/backend-routes.md]: There is a trailing comma missing after `name=""` which leads to `AnnotationException.php line 42: [Syntax Error] Expected Doctrine\Common\Annotations\DocLexer::T_CLOSE_PARENTHESIS, got 'defaults'` error

**Review by richardhj — CHANGES_REQUESTED** [review]

**Comment by richardhj** [comment]
Currently getting error
`The controller for URI "/contao/my-backend-route" is not callable. Controller "App\Controller\BackendController" cannot be fetched from the container because it is private. Did you forget to tag the service with "controller.service_arguments"?`

We should add the following to the services.yml (in this document).
```yml
  App\Controller\:
    resource: ../src/Controller/*
    tags:
      - name: controller.service_arguments
```
WDYT @aschempp?

**Comment by aschempp** [comment]
@richardhj that should not be necessary if your class extends the `AbstractController` and you have `autoconfig` enabled?

**Comment by fritzmg** [comment]
I personally prefer to _not_ extend from the `AbstractController` and instead making the service simply `public`.

But since the example does extend, no `services.yml` change should be necessary (if `autowire` is enabled).

File: docs/dev/guides/backend-routes.md (hunk 2 above)
Comment (fritzmg) [reply docs/dev/guides/backend-routes.md]: This still needs to be removed.

**Review by fritzmg — CHANGES_REQUESTED** [review]

**Comment by bytehead** [comment]
Why do you not prefer to extend from the `AbstractController`?

**Comment by fritzmg** [comment]
> Why do you not prefer to extend from the `AbstractController`?

I usually don't actually need any features from `AbstractController`.

**Review by fritzmg — APPROVED** [review]

**Comment by fritzmg** [comment]
Thank you @aschempp

**Pull request merged**